src/Microsoft.SqlTools.ServiceLayer/Utility/FileUtils.cs

```diff
@@ -8,6 +8,46 @@ namespace Microsoft.SqlTools.ServiceLayer.QueryExecution
 {
     internal static class FileUtils
     {
+        internal static string PeekDefinitionTempFolder = Path.GetTempPath() + "mssql_definition"; 
+        internal static bool PeekDefinitionTempFolderCreated = false;
+
+        internal static string GetPeekDefinitionTempFolder()
+        {
+            string tempPath;
+            if (!PeekDefinitionTempFolderCreated)
+            {
+                
+                try
+                {
+                    // create new temp folder
+                    string tempFolder = string.Format("{0}_{1}", FileUtils.PeekDefinitionTempFolder, DateTime.Now.ToString("yyyyMMddHHmmssffff"));
+                    DirectoryInfo tempScriptDirectory = Directory.CreateDirectory(tempFolder);
+                    FileUtils.PeekDefinitionTempFolder = tempPath = tempScriptDirectory.FullName;
+                    PeekDefinitionTempFolderCreated = true;
+                }
+                catch (Exception)
+                {
+                    // swallow exception and use temp folder to store scripts
+                    tempPath = Path.GetTempPath();
+                }
+            }
+            else
+            {
+                try
+                {
+                    // use tempDirectory name created previously
+                    DirectoryInfo tempScriptDirectory = Directory.CreateDirectory(FileUtils.PeekDefinitionTempFolder);
```

Comment:
If the path already exists, we want to just get the info for the path instead of create the folder anew, right? We can do that with `new DirectoryInfo(FileUtils.PeekDefinitionTempFolder)`